src/UI/Scripts/WorkingHours.js

```diff
@@ -0,0 +1,123 @@
+	const lengthForDayAndTime = 3;
+	const lengthForDayAndTimeAndBool = 4;
+	const indexForTimestampBool = 2;
+	const indexForDayAndTimeBool = 3;
+
+	const interfaceWeekdayTimestamps = workingHours.filter((x) => {
+		if(x.length === lengthForTimestampOnly) { // Find [timestamp, timestamp]
+			return true;
+		} else if(x.length === lengthForTimestampAndBool && typeof x[indexForTimestampBool] === "boolean") { // Find [timestamp, timestamp, openOrClosed]
+			return true;
+		}
+		return false;
+	});
+	const interfaceWeekdayDays = workingHours.filter((x) => {
+		if(x.length === lengthForDayAndTime) { // Find ["DayOfTheWeek", HH.MM, HH.MM]
+			return true;
+		} else if(x.length === lengthForDayAndTimeAndBool && typeof x[indexForDayAndTimeBool] === "boolean") { // Find ["DayOfTheWeek", HH.MM, HH.MM, openOrClosed]
+			return true;
+		}
+		return false;
+	});
+
+	let isOnline = handleTimestamps(interfaceWeekdayTimestamps, currentDate);
+	if(isOnline === null)
+		isOnline = handleDays(interfaceWeekdayDays, currentDate);
+
+	return isOnline;
+}
+
+/**
+ * Checks if the current time is within one of the open/close hours.
+ * Multiple ranges can be set for the same day.
+ * Check the return rules for exact rules used in the check.
+ *
+ * @param {Array<Number>} timestamps Format `[[openHoursTimestamp, closeHoursTimestamp],[..., ...]]`
+ * @param {Date} currentDate
+ * @return {Boolean|null}
+ * `TRUE` if current time is in between one of the start/end timestamps
+ * `FALSE` if current time is greater than the end timestamp AND we are on the same day as the end timestamp
+ * `NULL` if we haven't found any timestamp that matches the rules above
+ */
+function handleTimestamps(timestamps, currentDate) {
+	const indexForTimestampBool = 2;
+	const thousand = 1000;
+
+	let isOnline = null; // Return `null` if we haven't found any valid timestamps
+	for(let i = 0; i < timestamps.length; i++) {
+		const timestampInfo = timestamps[i];
+		const startDate = new Date(timestampInfo[0] * thousand);
+		const endDate = new Date(timestampInfo[1] * thousand);
+		const isOpen = timestampInfo[indexForTimestampBool] === undefined ? true : timestampInfo[indexForTimestampBool]; // Whether we are open (true) or closed (false) in this period
+
+		if(currentDate.getTime() > endDate.getTime() && currentDate.getDate() === endDate.getDate()) { // Return true if current time is between start and end
+			isOnline = !isOpen;
+			continue; // If there is another timestamp after this one, we give it the possibility to set us back to `online`
+		} else if(currentDate.getTime() >= startDate.getTime() && currentDate.getTime() <= endDate.getTime()) { // Return false if current time is after end, BUT only when current day is the same day as end
+			isOnline = Boolean(isOpen);
+			break; // We don't want another timestamp to set us back to `offline`
+		}
+	}
+
+	return isOnline;
+}
+
+function handleDays(days, currentDate) {
+	const ten = 10;
+
+	const currentDay = weekdays[currentDate.getDay()];
```

Comment:
Zelfde effect, maar zonder dat je zelf een weekdays array nodig hebt. 

```suggestion
	const currentDay = currentDate.toLocaleString('en-GB', { weekday: "long" }).toLowerCase();
```

Reply:
Done